pkg/trustedresources/verify.go

```diff
@@ -107,6 +107,19 @@ func VerifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.
 			Spec:       v.TaskSpec(),
 		}
 		return verifyResource(ctx, &task, k8s, signature, matchedPolicies)
+	case *v1.Task:
+		tm, signature, err := prepareObjectMeta(v.ObjectMeta)
+		if err != nil {
+			return VerificationResult{VerificationResultType: VerificationError, Err: err}
+		}
```

Comment:
This code chunk repeats in all 4 switch cases. We can change the function `prepareObjectMeta` 

from accepting struct
```go
func prepareObjectMeta(in metav1.ObjectMeta) (metav1.ObjectMeta, []byte, error) {
```
to accepting interface
```go
func prepareObjectMeta(in metav1.Object) (metav1.ObjectMeta, []byte, error) {
```

This way, we can avoid duplicate code by moving this out of switch statement.

Reply:
Oh thanks! Great suggestion